llvm/lib/Transforms/Instrumentation/NumericalStabilitySanitizer.cpp

```diff
@@ -634,10 +637,27 @@ NumericalStabilitySanitizer::NumericalStabilitySanitizer(Module &M)
         Attr, VoidTy, VTTy, VTTy, ShadowTy, ShadowTy, Int32Ty, Int1Ty, Int1Ty);
   }
 
-  NsanCopyValues = M.getOrInsertFunction("__nsan_copy_values", Attr, VoidTy,
-                                         PtrTy, PtrTy, IntptrTy);
-  NsanSetValueUnknown = M.getOrInsertFunction("__nsan_set_value_unknown", Attr,
-                                              VoidTy, PtrTy, IntptrTy);
+  NsanCopyFunction[0] =
```

Comment:
Since now we have multiple functions, arrays, and logic in GetInstrumentationCalleeIdxForMemOp, it'd be better to encapsulate them into helper classes, e.g., `class NsanCopy` and `class NsanSetUnknown`, to keep the internal details grouped together.

Reply:
Got it

Reply:
Done. I've tried to do it with one base class which implenents some basic virtual methods and a couple of derived class one per function family